src/test/java/lambda/part3/exercise/ReachIterableTest.java

```diff
@@ -0,0 +1,238 @@
+package lambda.part3.exercise;
+
+import data.Employee;
+import data.JobHistoryEntry;
+import data.Person;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Denis Verkhoturov, mod.satyr@gmail.com
+ */
+public class ReachIterableTest {
+    interface ReachIterable<T> {
+        boolean forNext(Consumer<T> consumer);
+
+        default ReachIterable<T> filter(final Predicate<T> predicate) {
+            return consumer -> this.forNext(
```

Comment:
Incorrect.
You should call `Consumer` in `forNext`. With this implementation you can return `true` withouc calling `Consumer`